### DISCUSSION TITLE
Home Assistant and Eenergy dashboard

### DISCUSSION THREAD

**Discussion by sotoniak** [mletenay/home-assistant-goodwe-inverter#55] [Q&A] [answered]
Hi, before I write my question.  This is great project. Respect! I was very surprised how many sensors GoodWe GW10K-ET has. Honestly, I don't understand most values :-)

And now my question. I fight with setting in Energy Home Assistant Dashboard. I have Inverter, battery and I don't know which values can I set. In select box with sensors many sensors miss. Please can anybody help me? 

https://www.home-assistant.io/blog/2021/08/04/home-energy-management/

**Comment by mfaust78** [comment, marked as answer]
Hi, I'm also having a GW10ET and played with the values for a few days. So far these settings look ok for me:
Grid Consumption -> GoodWe Meter Total Energy (import)
Return to Grid -> GoodWe Meter Total Energy (export)
Solar production -> GoodWe Total PV Generation
Battery System
- Energy going in to the battery -> GoodWe Total Battery Charge
- Energy coming out of the battery -> GoodWe Total Batter Discharge

Today looks like this for my system (with the above settings):
![image](https://user-images.githubusercontent.com/20776316/132059076-36c35969-90b5-4f80-99dc-1633f9ada20d.png)
![image](https://user-images.githubusercontent.com/20776316/132059172-5467fa54-2b71-4814-bca7-0a108642276f.png)

**Reply by mletenay** [reply]
This is exactly the setup I use too and is recommended in readme - https://github.com/mletenay/home-assistant-goodwe-inverter#home-assistant-energy-dashboard

**Reply by voodoopt** [reply]
In the ES model it's not that simple.

**Reply by JoLambinon** [reply]
Can you please explain how to get the values for batt charge and batt discharge with ES inverter ?

**Reply by voodoopt** [reply]
I use this code in sensor.yaml

# Template sensor for values of Batery charge (active_power < 0)
    battery_charge:
      friendly_name: "Battery charge"
      unique_id: battery_charge
      unit_of_measurement: 'W'
      value_template: >-
        {% if states('sensor.goodwe_pbattery')|float < 0 %}
          {{ states('sensor.goodwe_pbattery')|float * -1 }}
        {% else %}
          {{ 0 }}
        {% endif %}
# Template sensor for values of Batery discharge (active_power > 0)
    battery_discharge:
      friendly_name: "Battery discharge"
      unique_id: battery_discharge
      unit_of_measurement: 'W'
      value_template: >-
        {% if states('sensor.goodwe_pbattery')|float > 0 %}
          {{ states('sensor.goodwe_pbattery')|float }}
        {% else %}
          {{ 0 }}
        {% endif %}
# Sensor for Riemann Battery charge (W -> Wh)
  -- platform: integration
    source: sensor.battery_charge
    name: battery_charge_summ
    unit_prefix: k
    round: 1
    method: left
# Sensor for Riemann Battery discharge (W -> Wh)
  -- platform: integration
    source: sensor.battery_discharge
    name: battery_discharge_summ
    unit_prefix: k
    round: 1
    method: left

I don't know if it's 100% correct. But I am open to improvement. 
Come more opinions ;)

**Comment by iverlaek** [comment]
I'm also having the Goodwe-ET. It's running in parallel of a SolarEdge and an Enphase system. 

**Tip for a better user experience:**
I had to rename the Goodwe e_total sensor to Goodwe lifetime production (Similar name as used by both other inverters) 
![Screenshot_2 energy-dashboardpng](https://user-images.githubusercontent.com/89202964/141312796-b845f84e-490b-43fe-b6d7-0c47cd1bac2b.png)
for better consistancy and finding in the HA energy setup screen. 

e_total gives you a clue, but you have to know what to look for, which is with that many sensors a bit overwhelming. 

Please rename the sensor name so inexperienced users have this integration working immediately, and giving it a name more people will find. Also adding Goodwe to the names is helpful for distinguishing  if you are using multiple systems simultaneously.

**Comment by JoLambinon** [comment]
Thank you very much.  I will give it a try, and let you know.

Still wondering why these are not available from the API like the other sensors.

Sincerely

Jo Lambinon

Jo Lambinon
083 609 1286
***@***.******@***.***>



From: Ricardo AkA Voodoo ***@***.***>
Date: Wednesday, 17 November 2021 at 18:08
To: mletenay/home-assistant-goodwe-inverter ***@***.***>
Cc: JoLambinon ***@***.***>, Manual ***@***.***>
Subject: Re: [mletenay/home-assistant-goodwe-inverter] Home Assistant and Eenergy dashboard (Discussion #55)

I use this code in sensor.yaml

Template sensor for values of Batery charge (active_power < 0)

battery_charge:

  friendly_name: "Battery charge"

  unique_id: battery_charge

  unit_of_measurement: 'W'

  value_template: >-

    {% if states('sensor.goodwe_pbattery')|float < 0 %}

      {{ states('sensor.goodwe_pbattery')|float * -1 }}

    {% else %}

      {{ 0 }}

    {% endif %}

Template sensor for values of Batery discharge (active_power > 0)

battery_discharge:

  friendly_name: "Battery discharge"

  unique_id: battery_discharge

  unit_of_measurement: 'W'

  value_template: >-

    {% if states('sensor.goodwe_pbattery')|float > 0 %}

      {{ states('sensor.goodwe_pbattery')|float }}

    {% else %}

      {{ 0 }}

    {% endif %}

Sensor for Riemann Battery charge (W -> Wh)

-- platform: integration
source: sensor.battery_charge
name: battery_charge_summ
unit_prefix: k
round: 1
method: left

Sensor for Riemann Battery discharge (W -> Wh)

-- platform: integration
source: sensor.battery_discharge
name: battery_discharge_summ
unit_prefix: k
round: 1
method: left

I don't know if it's 100% correct. But I am open to improvement.
Come more opinions ;)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mletenay/home-assistant-goodwe-inverter/discussions/55#discussioncomment-1659097>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AKNJACAHFFBXKYEBAITBWYDUMPHPNANCNFSM5DMHVGSA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Reply by mletenay** [reply]
Because ES/EM inverters do not measure and provide those values. Only ET/EH do.

**Comment by JoLambinon** [comment]
Thank You for clarifying.  I understand.

Regards

Jo

Kind Regards
Jo Lambinon
083 609 1286
________________________________

**Comment by JoLambinon** [comment]
Hi Ricardo

How do you get the Riemann Battery charge and discharge sensors you created to show up in HA energy setup dropdown box please?

I now have these sensors but they are not available in the Energy configuration box for Home battery.

Regards

Jo



Jo Lambinon
083 609 1286
***@***.******@***.***>



From: Ricardo AkA Voodoo ***@***.***>
Date: Wednesday, 17 November 2021 at 18:08
To: mletenay/home-assistant-goodwe-inverter ***@***.***>
Cc: JoLambinon ***@***.***>, Manual ***@***.***>
Subject: Re: [mletenay/home-assistant-goodwe-inverter] Home Assistant and Eenergy dashboard (Discussion #55)

I use this code in sensor.yaml

Template sensor for values of Batery charge (active_power < 0)

battery_charge:

  friendly_name: "Battery charge"

  unique_id: battery_charge

  unit_of_measurement: 'W'

  value_template: >-

    {% if states('sensor.goodwe_pbattery')|float < 0 %}

      {{ states('sensor.goodwe_pbattery')|float * -1 }}

    {% else %}

      {{ 0 }}

    {% endif %}

Template sensor for values of Batery discharge (active_power > 0)

battery_discharge:

  friendly_name: "Battery discharge"

  unique_id: battery_discharge

  unit_of_measurement: 'W'

  value_template: >-

    {% if states('sensor.goodwe_pbattery')|float > 0 %}

      {{ states('sensor.goodwe_pbattery')|float }}

    {% else %}

      {{ 0 }}

    {% endif %}

Sensor for Riemann Battery charge (W -> Wh)

-- platform: integration
source: sensor.battery_charge
name: battery_charge_summ
unit_prefix: k
round: 1
method: left

Sensor for Riemann Battery discharge (W -> Wh)

-- platform: integration
source: sensor.battery_discharge
name: battery_discharge_summ
unit_prefix: k
round: 1
method: left

I don't know if it's 100% correct. But I am open to improvement.
Come more opinions ;)

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub<https://github.com/mletenay/home-assistant-goodwe-inverter/discussions/55#discussioncomment-1659097>, or unsubscribe<https://github.com/notifications/unsubscribe-auth/AKNJACAHFFBXKYEBAITBWYDUMPHPNANCNFSM5DMHVGSA>.
Triage notifications on the go with GitHub Mobile for iOS<https://apps.apple.com/app/apple-store/id1477376905?ct=notification-email&mt=8&pt=524675> or Android<https://play.google.com/store/apps/details?id=com.github.android&referrer=utm_campaign%3Dnotification-email%26utm_medium%3Demail%26utm_source%3Dgithub>.

**Comment by Xfire207** [comment]
can und please help how i do this Sensors for charge/ discharge for EM inverter please . i have Download with hac goodwe "Experiment" but the sensor is not including. 

greetz from Germany 

Pocmartin207@gmx.de

**Comment by mletenay** [comment]
ES/EM inverters do not provide these cumulative values, you need to calculated them in HA. See https://github.com/mletenay/home-assistant-goodwe-inverter#cumulative-energy-values
